go/vt/discovery/healthcheck.go

```diff
@@ -349,106 +349,74 @@ func (hc *HealthCheckImpl) checkConn(hcc *healthCheckConn, name string) {
 	}
 	hc.initialUpdatesWG.Done()
 
-	// retry health check if it fails
+	// Read stream health responses.
 	for {
-		// Try to connect to the tablet.
-		stream, err := hcc.connect(hc)
+		err := hcc.stream(hc, func(shr *querypb.StreamHealthResponse) error {
+			return hcc.processResponse(hc, shr)
+		})
 		if err != nil {
-			select {
-			case <-hcc.ctx.Done():
-				return
-			default:
-			}
 			hcc.mu.Lock()
-			hcc.tabletStats.Serving = false
-			hcc.tabletStats.LastError = err
-			target := hcc.tabletStats.Target
+			ts := hcc.tabletStats
 			hcc.mu.Unlock()
-			hcErrorCounters.Add([]string{target.Keyspace, target.Shard, topoproto.TabletTypeLString(target.TabletType)}, 1)
-
-			// Sleep until the next retry is up or the context is done/canceled.
-			select {
-			case <-hcc.ctx.Done():
-			case <-time.After(hc.retryDelay):
+			// notify downstream for serving status change
+			if hc.listener != nil {
+				hc.listener.StatsUpdate(&ts)
```

Comment:
Not too familiar with this code. Tests are still passing after the delete. I actually think this can be further simplified, but I don't want to lose focus of my main objective.